core/src/main/java/org/elasticsearch/cluster/ClusterStateUpdateTask.java

```diff
@@ -20,46 +20,42 @@
 package org.elasticsearch.cluster;
 
 import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Priority;
 import org.elasticsearch.common.unit.TimeValue;
-import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+
+import java.util.List;
 
 /**
  * A task that can update the cluster state.
  */
-abstract public class ClusterStateUpdateTask {
+abstract public class ClusterStateUpdateTask implements ClusterStateTaskConfig, ClusterStateTaskExecutor<Object>, ClusterStateTaskListener {
```

Comment:
@bleskes Because of the change in e4c6bd4a12ef824a2944e85b1d2b0f56adf9addf to more explicitly associate tasks with their execution results, tasks that pass through `ClusterService#submitStateUpdateTask(String, ClusterStateUpdateTask)` and previously delegated to ` ClusterService#submitStateUpdateTask(String, Object, config, executor, listener)` by passing `null` for the task now need to pass something not null. As `Void` is uninstantiable, I changed it to `Object`. However, I'm thinking that c19c30b7a8e4315f61900b8854cf7dbba9c62d2f might be better. What do you think?